build/lib/pgBackRestBuild/Config/Data.pm

```diff
@@ -2260,6 +2268,57 @@ my %hConfigDefine =
         ],
     },
 
+    &CFGOPT_REPO_GCS_BUCKET =>
+    {
+        &CFGDEF_GROUP => CFGOPTGRP_REPO,
+        &CFGDEF_SECTION => CFGDEF_SECTION_GLOBAL,
+        &CFGDEF_TYPE => CFGDEF_TYPE_STRING,
+        &CFGDEF_REQUIRED => true,
+        &CFGDEF_DEPEND =>
+        {
+            &CFGDEF_DEPEND_OPTION => CFGOPT_REPO_GCS_KEY_TYPE,
+        },
+        &CFGDEF_COMMAND => CFGOPT_REPO_TYPE,
+    },
+
+    &CFGOPT_REPO_GCS_ENDPOINT =>
+    {
+        &CFGDEF_GROUP => CFGOPTGRP_REPO,
+        &CFGDEF_SECTION => CFGDEF_SECTION_GLOBAL,
+        &CFGDEF_TYPE => CFGDEF_TYPE_STRING,
+        &CFGDEF_DEFAULT => 'storage.googleapis.com',
+        &CFGDEF_DEPEND => CFGOPT_REPO_GCS_BUCKET,
+        &CFGDEF_COMMAND => CFGOPT_REPO_TYPE,
+    },
+
+    &CFGOPT_REPO_GCS_KEY =>
+    {
+        &CFGDEF_GROUP => CFGOPTGRP_REPO,
+        &CFGDEF_SECTION => CFGDEF_SECTION_GLOBAL,
+        &CFGDEF_TYPE => CFGDEF_TYPE_STRING,
+        &CFGDEF_DEPEND => CFGOPT_REPO_GCS_BUCKET,
+        &CFGDEF_COMMAND => CFGOPT_REPO_TYPE,
+    },
+
+    &CFGOPT_REPO_GCS_KEY_TYPE =>
+    {
+        &CFGDEF_GROUP => CFGOPTGRP_REPO,
+        &CFGDEF_SECTION => CFGDEF_SECTION_GLOBAL,
+        &CFGDEF_TYPE => CFGDEF_TYPE_STRING,
+        &CFGDEF_DEFAULT => 'service',
+        &CFGDEF_ALLOW_LIST =>
+        [
+            'service',
+            'token',
+        ],
+        &CFGDEF_DEPEND =>
+        {
+            &CFGDEF_DEPEND_OPTION => CFGOPT_REPO_TYPE,
+            &CFGDEF_DEPEND_LIST => [CFGOPTVAL_REPO_TYPE_GCS],
+        },
+        &CFGDEF_COMMAND => CFGOPT_REPO_TYPE,
+    },
```

Comment:
Just to confirm: so none of the GCS options would need `CFGDEF_SECURE => true` - right?

Reply:
Hmmm, no. gcs-key needs to be secure. It will usually be a filename bit it could be a token, and there are other possibilities in the future. Fixed in 399eb818af02db4be5ffa.